app/javascript/src/utils/compiler.js

```diff
@@ -137,6 +138,93 @@ function extractAndInsertMixins(joinedItems) {
   return joinedItems
 }
 
+/**
+ * @param {string} source
+ * @param {number} fromIndex
+ * @returns
+ */
```

Comment:
This is nice and all, but we don't do it anywhere else so let's not do it here either

Reply:
I used these mainly for myself since my IDE wouldn't recognize types and show autocomplete otherwise. Do these work on your IDE?

I'd argue JSDocs are a necessity for non TypeScript projects. What if instead we start rolling these out gradually as we change things?

Reply:
I personally haven't had an issue with it, but I can definitely see it being useful. In any case, I'd prefer to leave that for a different PR if we decide to do it at all